include/zephyr/drivers/auxdisplay.h

```diff
@@ -0,0 +1,824 @@
+ */
+typedef int (*auxdisplay_brightness_set_api)(const struct device *dev,
+					     uint8_t brightness);
+
+/**
+ * @typedef	auxdisplay_backlight_get_api
+ * @brief	Callback API to get the current and minimum/maximum supported
+ *		backlight settings of the display
+ * See auxdisplay_backlight_set() for argument description
+ */
+typedef int (*auxdisplay_backlight_get_api)(const struct device *dev,
+					    uint8_t *backlight);
+
+/**
+ * @typedef	auxdisplay_backlight_set_api
+ * @brief	Callback API to set the backlight status
+ * See auxdisplay_backlight_set() for argument description
+ */
+typedef int (*auxdisplay_backlight_set_api)(const struct device *dev,
+					    uint8_t backlight);
+
+/**
+ * @typedef	auxdisplay_is_busy_api
+ * @brief	Callback API to check if the display is busy with an operation
+ * See auxdisplay_is_busy() for argument description
+ */
+typedef int (*auxdisplay_is_busy_api)(const struct device *dev);
+
+/**
+ * @typedef	auxdisplay_custom_character_set_api
+ * @brief	Callback API to set a customer character on the display for usage
+ * See auxdisplay_custom_character_set() for argument description
+ */
+typedef int (*auxdisplay_custom_character_set_api)(const struct device *dev,
+						   struct auxdisplay_character *character);
+
+/**
+ * @typedef	auxdisplay_write_api
+ * @brief	Callback API to write text to the display
+ * See auxdisplay_write() for argument description
+ */
+typedef int (*auxdisplay_write_api)(const struct device *dev,
+				    const uint8_t *data, uint16_t len);
+
+/**
+ * @typedef	auxdisplay_custom_command_api
+ * @brief	Callback API to send a custom command to the display
+ * See auxdisplay_custom_command() for argument description
+ */
+typedef int (*auxdisplay_custom_command_api)(const struct device *dev,
+					     struct auxdisplay_custom_data *command);
+
+/**
+ * @brief Text display driver API
+ * API which a textual display driver should expose
+ */
+struct auxdisplay_driver_api {
+	auxdisplay_cursor_set_enabled_api cursor_set_enabled;
+	auxdisplay_position_blinking_set_enabled_api position_blinking_set_enabled;
+	auxdisplay_cursor_shift_set_api cursor_shift_set;
+	auxdisplay_cursor_position_set_api cursor_position_set;
+	auxdisplay_cursor_position_get_api cursor_position_get;
+	auxdisplay_display_position_set_api display_position_set;
+	auxdisplay_display_position_get_api display_position_get;
+	auxdisplay_capabilities_get_api capabilities_get;
+	auxdisplay_clear_api clear;
+	auxdisplay_brightness_get_api brightness_get;
+	auxdisplay_brightness_set_api brightness_set;
+	auxdisplay_backlight_get_api backlight_get;
+	auxdisplay_backlight_set_api backlight_set;
+	auxdisplay_is_busy_api is_busy;
+	auxdisplay_custom_character_set_api custom_character_set;
+	auxdisplay_write_api write;
+	auxdisplay_custom_command_api custom_command;
+};
+
+/**
+ * @brief		Set cursor enabled status on a text display
+ *
+ * @param dev		Pointer to device structure
```

Comment:
Description is ambiguous here, maybe "Aux display device instance"

Reply:
Updated